website/data-storage/src/save.spec.ts

```diff
@@ -0,0 +1,128 @@
+import { Test } from '@nestjs/testing';
+import { getRepositoryToken } from '@nestjs/typeorm';
+
+import { Meta } from './metadata/entities/meta.entity';
+import { Program } from './programs/entities/program.entity';
+import { ProgramsService } from './programs/programs.service';
+
+describe('Programs Service', () => {
+  let programsService!: ProgramsService;
+
+  const Meta_create = jest.fn(() => ({ id: 7357 }));
+  const Meta_save = jest.fn(async () => ({ id: 7357 }));
+  const Meta_findAndCount = jest.fn(async () => [[{ id: 7357 }], 1]);
+  const Meta_findOne = jest.fn(async () => ({ id: 7357 }));
+
+  const Program_create = jest.fn(() => ({ id: 7357 }));
+  const Program_save = jest.fn(async () => ({ id: 7357 }));
+  const Program_findAndCount = jest.fn(async () => [[{ id: 7357 }], 1]);
+  const Program_findOne = jest.fn(async () => ({ id: 7357 }));
+
+  beforeAll(async () => {
+    const moduleRef = await Test.createTestingModule({
+      providers: [
+        // Stub upstream repos:
+        {
+          provide: getRepositoryToken(Program),
+          useValue: {
+            create: Program_create,
+            save: Program_save,
+            findAndCount: Program_findAndCount,
+            findOne: Program_findOne,
+          },
+        },
+        {
+          provide: getRepositoryToken(Meta),
+          useValue: {
+            create: Meta_create,
+            save: Meta_save,
+            findAndCount: Meta_findAndCount,
+            findOne: Meta_findOne,
+          },
+        },
+        // And use the default Programs service as it is:
+        ProgramsService,
+      ],
+    }).compile();
+
+    programsService = moduleRef.get(ProgramsService);
+  });
+
+  afterEach(() => {
+    for (const { mockReset } of [
+      Meta_create,
+      Meta_save,
+      Meta_findAndCount,
+      Meta_findOne,
+      Program_create,
+      Program_save,
+      Program_findAndCount,
+      Program_findOne,
+    ]) {
+      mockReset();
+    }
+  });
+
+  it('should record uploaded program', async () => {
+    // Given an initialized Programs Service,
+    // when:
+    await programsService.save({
+      // This number spells "test" if you squint hard enough.
+      id: 7357,
```

Comment:
Actually id is a string in the form of hex

Reply:
Understood; now it is in tests too.